<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

  <xacro:macro name="robot6dof_system_position_only" params="name prefix">

    <ros2_control name="${name}" type="system">
      <hardware>
        <classType>ros2_control_demo_driver/Robot6DofSystemHardware</classType>
        <param name="example_param_hw_start_duration_sec">2.0</param>
        <param name="example_param_hw_stop_duration_sec">3.0</param>
        <param name="example_param_hw_slowdown">2.0</param>
      </hardware>
      <joint name="${prefix}joint_1">
        <classType>ros2_control_components/PositionJoint</classType>
        <param name="min">-1</param>
        <param name="max">1</param>
      </joint>
      <joint name="${prefix}joint_2">
        <classType>ros2_control_components/PositionJoint</classType>
        <param name="min">-1</param>
        <param name="max">1</param>
	  </joint>
      <joint name="${prefix}joint_3">
        <classType>ros2_control_components/PositionJoint</classType>
        <param name="min">-1</param>
        <param name="max">1</param>
	  </joint>
      <joint name="${prefix}joint_4">
        <classType>ros2_control_components/PositionJoint</classType>
        <param name="min">-1</param>
        <param name="max">1</param>
      </joint>
      <joint name="${prefix}joint_5">
        <classType>ros2_control_components/PositionJoint</classType>
        <param name="min">-1</param>
        <param name="max">1</param>
      </joint>
      <joint name="${prefix}joint_6">
        <classType>ros2_control_components/PositionJoint</classType>
        <param name="min">-1</param>
        <param name="max">1</param>
      </joint>
    </ros2_control>

  </xacro:macro>

</robot>

